### PR TITLE
refactor: use async function and support egg@2

### PR DESCRIPTION
**Pull request by dead-horse**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

**Comment by dead-horse** [comment]
@eggjs/core

**Review by fengmk2 — APPROVED** [review]

File: lib/view.js [hunk 1]
```diff
@@ -23,16 +23,17 @@ module.exports = class EjsView {
...
-    const html = yield this[RENDER](filename, locals, config);
+    const html = await this[RENDER](filename, locals, config);
     if (!config.layout) {
       return html;
     }
 
     locals.body = html;
-    const layout = yield this.app.view.resolve(config.layout);
-    return yield this[RENDER](layout, locals, config);
+    console.log(this.app.view.resolve.toString());
```
Comment (atian25) [inline lib/view.js]: remove

**Review by atian25 — APPROVED** [review]

**Review by atian25 — APPROVED** [review]

**Comment by atian25** [comment]
你发

**Comment by dead-horse** [comment]
@atian25 加一下权限

**Comment by atian25** [comment]
```bash
➜  npm owner ls egg-view-ejs
dead_horse <dead_horse@qq.com>
dead-horse <dead_horse@qq.com>
fengmk2 <fengmk2@gmail.com>
popomore <sakura9515@gmail.com>
atian25 <atian25@qq.com>
cycok <342930982@qq.com>
```

不要问我为什么加了两个~

**Pull request merged**